node/forest_libp2p/src/blocksync/message.rs

```diff
@@ -2,70 +2,26 @@
 // SPDX-License-Identifier: Apache-2.0, MIT
 
 use forest_blocks::{Block, BlockHeader, FullTipset, Tipset, BLOCK_MESSAGE_LIMIT};
-use forest_cid::Cid;
 use forest_encoding::tuple::*;
 use forest_message::{SignedMessage, UnsignedMessage};
 use std::convert::TryFrom;
 
-/// Blocksync request options
-pub const BLOCKS: u64 = 1;
-pub const MESSAGES: u64 = 2;
-
-/// The payload that gets sent to another node to request for blocks and messages. It get DagCBOR serialized before sending over the wire.
-#[derive(Clone, Debug, PartialEq, Serialize_tuple, Deserialize_tuple)]
-pub struct BlockSyncRequest {
-    /// The tipset to start sync from
-    pub start: Vec<Cid>,
-    /// The amount of epochs to sync by
-    pub request_len: u64,
-    /// 1 = Block only, 2 = Messages only, 3 = Blocks and Messages
-    pub options: u64,
-}
-
-/// The response to a BlockSync request.
-#[derive(Clone, Debug, PartialEq, Serialize_tuple, Deserialize_tuple)]
-pub struct BlockSyncResponse {
-    /// Error code
-    pub status: u64,
-    /// Status message indicating failure reason
-    pub message: String,
-    /// The tipsets requested
-    pub chain: Vec<TipsetBundle>,
-}
-
-impl BlockSyncResponse {
-    /// Converts blocksync response into result.
-    /// Returns an error if the response status is not `Ok`.
-    /// Tipset bundle is converted into generic return type with `TryFrom` trait impl.
-    pub fn into_result<T>(self) -> Result<Vec<T>, String>
-    where
-        T: TryFrom<TipsetBundle, Error = String>,
-    {
-        if self.status != 0 {
-            // TODO implement a better error type than string if needed to be handled differently
-            return Err(format!("Status {}: {}", self.status, self.message));
-        }
-
-        self.chain.into_iter().map(T::try_from).collect()
-    }
-}
-
```

Comment:
Why did these types need to be moved? Makes more sense for these to be here (also to be symmetric with `hello` protocol message definitions)

Reply:
This was my attempt to justify the creation of a new struct and module (`provider.rs`) by relocating these how I thought relevant structs over there. See my comment above regarding the channel/function idea.

Reply:
Can you just keep these types where they were and just have the provider logic in that new file? There should be no issue accessing, they are all in the same crate

Reply:
sure, moved them back.